### ISSUE TITLE
[BUG] Fail to build on macos: explicit instantiation of 'deepmd::Region' must occur in namespace 'deepmd'

### ISSUE THREAD

**Issue by njzjz** [deepmodeling/deepmd-kit#1070]
**Summary**

See https://github.com/conda-forge/deepmd-kit-feedstock/pull/23. The error is raised when building `region.cc`

```
2021-08-29T14:13:21.1799820Z   [ 27%] Building CXX object lib/CMakeFiles/deepmd.dir/src/region.cc.o
2021-08-29T14:13:21.7569440Z   /tmp/pip-req-build-5uynhvg7/source/lib/src/region.cc:25:17: error: explicit instantiation of 'deepmd::Region' must occur in namespace 'deepmd'
2021-08-29T14:13:21.7606880Z   template struct Region<double>;
2021-08-29T14:13:21.7695530Z                   ^
2021-08-29T14:13:21.7773140Z   /tmp/pip-req-build-5uynhvg7/source/lib/include/region.h:6:8: note: explicit instantiation refers here
2021-08-29T14:13:21.7774390Z   struct Region
2021-08-29T14:13:21.7775380Z          ^
2021-08-29T14:13:21.7777130Z   /tmp/pip-req-build-5uynhvg7/source/lib/src/region.cc:26:17: error: explicit instantiation of 'deepmd::Region' must occur in namespace 'deepmd'
2021-08-29T14:13:21.7778390Z   template struct Region<float>;
2021-08-29T14:13:21.7779450Z                   ^
2021-08-29T14:13:21.7781030Z   /tmp/pip-req-build-5uynhvg7/source/lib/include/region.h:6:8: note: explicit instantiation refers here
2021-08-29T14:13:21.7782140Z   struct Region
2021-08-29T14:13:21.7783050Z          ^
2021-08-29T14:13:21.7947800Z   2 errors generated.
```

Related codes:
https://github.com/deepmodeling/deepmd-kit/blob/1a25414dcd18b12316250baa330b690f3f5510fe/source/lib/src/region.cc#L9-L26

https://github.com/deepmodeling/deepmd-kit/blob/1a25414dcd18b12316250baa330b690f3f5510fe/source/lib/include/region.h#L5-L12

**Deepmd-kit version, installation way, input file, running commands, error log, etc.**

source code of 2.0.0

**Steps to Reproduce**

on macos 10.15.7, clang 11.1

```sh
pip install .
```

**Further Information, Files, and Links**

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=369278&view=logs&jobId=a8dee1cf-d17f-5cb0-b53f-4a93dfdbfb0c&j=a8dee1cf-d17f-5cb0-b53f-4a93dfdbfb0c&t=d9dbf137-9a7f-5666-c60c-9e1b2fc31445
